### ISSUE TITLE
Access violation in module libmariadb.dll

### ISSUE THREAD

**Issue by Leon-001** [HeidiSQL/HeidiSQL#809]
## Steps to reproduce this issue

1. connect to server 
- with SSL enabled
- running through wine on Linux (kubuntu)

## Current behavior
- a window opens with "EAccessViolation" (see attached file)
[bugreport.txt](https://github.com/HeidiSQL/HeidiSQL/files/3875498/bugreport.txt)

## Expected behavior

- connection to server established

## Possible solution

- connecting to a local server without SSL enabled works fine
- the error did not occur with the previous version of HeidiSQL (9.5?)

- the error does also occur with the latest nightly build (10.2.0.5757)
(if I choose Library = libmariadb.dll)
-> but not, if I choose Library = libmysql.dll !! :-)

## Environment

- wine version 4.0.2
- I have tried these Windows Versions: 7, 8.1, 10

- uname -a gives:
Linux  4.15.0-70-generic #79-Ubuntu x86_64 x86_64 x86_64 GNU/Linux

* **HeidiSQL version:** 
10.2.0.5599

  
* **Database system and version:** 
mysql --version gives:
  Ver 15.1 Distrib 10.1.41-MariaDB, for debian-linux-gnu (x86_64) using readline 5.2

**Comment by ansgarbecker** [comment]
5599  is not the latest build. Please update to the latest one and try again. Not sure if that works on Wine, but normally you can do that through _Help > Check for updates_

**Comment by Leon-001** [comment]
Hi Ansgar,
thanks for the quick response and the great software!
Version 5599 has the error. I have installed the nightly build, and that works somehow (see above). But as it is not recommended to use in production, I would prefer to use a stable version :)

**Comment by ansgarbecker** [comment]
> I would prefer to use a stable version

Yes of course! Just for reporting bugs the latest builds are a must, to prevent duplicate bug reports which were already fixed.

Thanks for your quick feedback!

**Comment by Leon-001** [comment]
Hi,
ok. Should I then open a new report on this issue? Because I would expect to be able to connect to MariaDB server with Library = libmariadb.dll
But this is not possible for me, even with Version 5757.

**Comment by ansgarbecker** [comment]
Ah, I obviously misinterpreted your
> I have installed the nightly build, and that works somehow
... with "that fixed the issue" :)

I'll reopen this one.

Please just upload a crash report with that new build - it's potentially different from the one you uploaded above.

**Comment by Leon-001** [comment]
ok, sorry for being not clear. 
Here comes the new file: [bugreport1.txt](https://github.com/HeidiSQL/HeidiSQL/files/3881150/bugreport1.txt)
I hope, that does not matter, but I have upgraded my operating system this morning!
If you like, I could post more details, or open a new report...

**Comment by ansgarbecker** [comment]
Found a forum thread describing a similar issue: https://www.heidisql.com/forum.php?t=26639

Did you already try a newer 4.x or older 3.x version of Wine?

**Comment by Leon-001** [comment]
I have tried it with wine-4.20 (staging) now. But I still get the error: [bugreport2.txt](https://github.com/HeidiSQL/HeidiSQL/files/3882220/bugreport2.txt)

**Comment by ansgarbecker** [comment]
Hm, that new bugreport does not even contain a crash message, just "access violation", and no callstack.

Could you please ensure you have the latest `libmariadb.dll`, by downloading the appropriate one from https://github.com/HeidiSQL/HeidiSQL/tree/master/out (depending on 32/64bit you are using)

**Comment by Leon-001** [comment]
oh, it seems, that I have indeed used an old version!
With the new one, Heidi does not crash anymore. But instead I get the following error message:
"lost connection to mysql server at 'sending authentication information' system error: 0"
![heidi-message](https://user-images.githubusercontent.com/21216698/69479342-5cea4c80-0dfc-11ea-853d-6bf1cf31a0bf.png)
I have also tried the other files:
libmariadb64.dll: not working (as expected)
libmysql32.dll: working (as before)

**Comment by ansgarbecker** [comment]
Probably related to #589

**Comment by ansgarbecker** [comment]
Hey @Leon-001 I'm just looking at older tickets right now, and I think I will close this one. I would recommend to use the new native Linux version instead, which should probably not be so error prone.
Thank you!
